### ISSUE TITLE
ValidationError: '5 is greater than the maximum of -180' when requesting a cube with spatial subset for several datasets

### ISSUE THREAD

**Issue by AliceBalfanz** [dcs4cop/xcube-cds#28]
When requesting a cds cube with a spatial subset, an Validation Error occurs for some dataset ids:  `ValidationError: '5 is greater than the maximum of -180'`
this happens for the following dataset ids: 

- satellite-soil-moisture:saturation:daily
- satellite-soil-moisture:saturation:10-day
- satellite-soil-moisture:saturation:monthly
- satellite-soil-moisture:volumetric:daily
- satellite-soil-moisture:volumetric:10-day
- satellite-soil-moisture:volumetric:monthly

to reproduce: 

```
from xcube.core.store import find_data_store_extensions
from xcube.core.store import get_data_store_params_schema
from xcube.core.store import new_data_store

store = new_data_store('cds')

dataset = store.open_data('satellite-soil-moisture:saturation:daily', 
                          variable_names=['soil_moisture_saturation'], 
                          bbox=[5, 35, 6, 36], 
                          spatial_res=0.25, 
                          time_range=['1978-11-01', '1978-11-06'])
dataset
```

The stacktrace: 

``` 
---------------------------------------------------------------------------
ValidationError                           Traceback (most recent call last)
<ipython-input-6-82c0cb0543ee> in <module>
      3                           bbox=[5, 35, 6, 36],
      4                           spatial_res=0.25,
----> 5                           time_range=['1978-11-01', '1978-11-06'])
      6 dataset

~/Desktop/projects/xcube-cds/xcube_cds/store.py in open_data(self, data_id, opener_id, **open_params)
    768         self._assert_valid_opener_id(opener_id)
    769         self._validate_data_id(data_id)
--> 770         return super().open_data(data_id, **open_params)
    771 
    772     ###########################################################################

~/Desktop/projects/xcube-cds/xcube_cds/store.py in open_data(self, data_id, **open_params)
    417 
    418         schema = self.get_open_data_params_schema(data_id)
--> 419         schema.validate_instance(open_params)
    420         handler = self._handler_registry[data_id]
    421 

~/Desktop/projects/xcube/xcube/util/jsonschema.py in validate_instance(self, instance)
    101                             format_checker=jsonschema.draft7_format_checker,
    102                             # We have to explicitly sanction tuples as arrays.
--> 103                             types=dict(array=(list, tuple)))
    104 
    105     def to_instance(self, value: Any) -> Any:

~/miniconda3/envs/xcube/lib/python3.7/site-packages/jsonschema/validators.py in validate(instance, schema, cls, *args, **kwargs)
    932     error = exceptions.best_match(validator.iter_errors(instance))
    933     if error is not None:
--> 934         raise error
    935 
    936 

ValidationError: 5 is greater than the maximum of -180

Failed validating 'maximum' in schema['properties']['bbox']['items'][0]:
    {'maximum': -180, 'minimum': -180, 'type': 'number'}

On instance['bbox'][0]:
    5
```

**Comment by pont-us** [comment]
Subsetting is not supported by the soil moisture dataset, so it's expected that the call will fail. The error message is correct (since the min-x value of the bounding box has a maximum permitted value of -180), but admittedly a bit cryptic. It could be made more explicit that [-180, -90, 180, 90] (i.e. the whole world) the only permitted value for the bbox parameter.

**Comment by pont-us** [comment]
After further discussion, I'm implementing subsetting within the CDS store, so that smaller bounding boxes will be allowed (though of course the full data still has to be fetched via the backend, so no time or bandwidth is saved).

**Comment by pont-us** [comment]
PR #37 (addressing Issue #35) has removed the store-internal subsetting again, but I'm keeping this issue closed because PR #38 technically fixes the problem (by disallowing the bbox parameter) and the latest xcube gen2 improvements actually fix the problem (by incorporating general-purpose subsetting).
